pkg/controllers/nodeclass/validation.go

```diff
@@ -40,6 +60,137 @@ func (n Validation) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (
 			fmt.Sprintf("%q tag does not pass tag validation requirements", offendingTag))
 		return reconcile.Result{}, reconcile.TerminalError(fmt.Errorf("%q tag does not pass tag validation requirements", offendingTag))
 	}
+	// Auth Validation
+	if nodeClass.StatusConditions().Get(v1.ConditionTypeSecurityGroupsReady).IsFalse() || nodeClass.StatusConditions().Get(v1.ConditionTypeAMIsReady).IsFalse() || nodeClass.StatusConditions().Get(v1.ConditionTypeInstanceProfileReady).IsFalse() || nodeClass.StatusConditions().Get(v1.ConditionTypeSubnetsReady).IsFalse() {
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "DependenciesNotReady", "Waiting for SecurityGroups, AMIs, and InstanceProfiles")
+		// nolint:nilerr
+		return reconcile.Result{}, nil
+	}
+	nodeClaim := &karpv1.NodeClaim{
+		Spec: karpv1.NodeClaimSpec{
+			NodeClassRef: &karpv1.NodeClassReference{
+				Name: nodeClass.ObjectMeta.Name,
+			},
+		},
+	}
+	tags, err := utils.GetTags(nodeClass, nodeClaim, options.FromContext(ctx).ClusterName)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("getting tags, %w", err)
+	}
+
+	createFleetInput := instance.GetCreateFleetInput(nodeClass, string(karpv1.CapacityTypeOnDemand), tags, mockLaunchTemplateConfig())
+	createFleetInput.DryRun = aws.Bool(true)
+
+	if _, err := n.ec2api.CreateFleet(ctx, createFleetInput); awserrors.IgnoreDryRunError(err) != nil {
+		log.FromContext(ctx).Info(err.Error())
+		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeValidationSucceeded, "CreateFleetAuthCheckFailed", "Unauthorized Operation")
+		// nolint:nilerr
```

Comment:
To bolster my statement above, you don't need this `nolint` if you don't ignore unexpected errors